core/trigonometry.cpp

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#include <cmath>
+
+#include "scipp/common/constants.h"
+#include "scipp/core/except.h"
+#include "scipp/core/transform.h"
+#include "scipp/core/variable.h"
+
+namespace scipp::core {
+
+struct no_variance {
+  template <class T>
+  scipp::core::detail::ValueAndVariance<T>
+  operator()(const scipp::core::detail::ValueAndVariance<T> &) const {
+    // TODO A better way for disabling operations with variances is needed.
+    throw std::runtime_error("Operation does not support data with variances.");
+  }
+};
+
+Variable to_rad(Variable var) {
+  const auto scale =
+      makeVariable<double>({}, units::rad / units::deg, {pi<double> / 180.0});
+  return var * scale;
+}
+
+Variable sin(const Variable &var) {
+  using std::sin;
+  const Variable &rad = var.unit() == units::deg ? to_rad(var) : var;
```

Comment:
You could also move this check inside `to_rad` which would then either convert if you have `deg` as input or simply return the original if input is `rad`. Then you would simply have
`const Variable &rad = to_rad(var)`

Reply:
I did not do that because that would always force a copy, even if no conversion is required.